### PR TITLE
Feature/discourse link fix

### PR DESCRIPTION
**Pull request by agduncan94**
We will now redirect any attempts to get to /tools to /containers (when looking at an individual tool) and also deal with redirecting to the decoded URL.

https://github.com/ga4gh/dockstore/issues/1290

### CODE REVIEW

File: src/app/container/container.component.ts [hunk 1]
```diff
@@ -303,4 +316,15 @@ export class ContainerComponent extends Entry {
...
 
+   /**
+    * Will change the /tools in the current URL with /containers
+    * @return {void}
+    */
+   switchToolsToContainers(): void {
+     const url = window.location.href.replace('/tools', '/containers');
+     const toolsIndex = window.location.href.indexOf('/tools');
+     const newPath = url.substring(toolsIndex);
+     console.log(newPath);
```
Comment (garyluu) [inline src/app/container/container.component.ts]: Probably don't need this

File: src/app/workflow/workflow.component.ts [hunk 2]
```diff
@@ -71,17 +71,23 @@ export class WorkflowComponent extends Entry {
...
 
-    // Initialize discourse urls
-    (<any>window).DiscourseEmbed = {
-      discourseUrl: Dockstore.DISCOURSE_URL,
-      discourseEmbedUrl: decodeURIComponent(trimmedURL)
-    };
+      // Decode the URL
+      this.decodeURL(this._toolType);
+
+      const indexOfLastColon = window.location.href.indexOf(':', window.location.href.indexOf('workflows'));
```
Comment (garyluu) [inline src/app/workflow/workflow.component.ts]: In general, lots of code looks similar to each other.  throw it into entry.ts when possible

**Comment by agduncan94** [comment]
Assuming this passes the test it is ready for review.

**Comment by agduncan94** [comment]
Now it is ready for review.

File: cypress/integration/toolDetails.js [hunk 3]
```diff
@@ -1,11 +1,43 @@
+describe('Variations of URL', function() {
```
Comment (denis-yuen) [inline cypress/integration/toolDetails.js]: :+1:

File: cypress/integration/workflowDetails.js [hunk 4]
```diff
@@ -1,13 +1,35 @@
+describe('Variations of URL', function() {
```
Comment (denis-yuen) [inline cypress/integration/workflowDetails.js]: I may refer to these two as a FAQ entry actually, might be worth telling people what we mean by "canonical URL."

**Pull request merged**